### PR TITLE
OCPNODE-2023: add metrics_host to config

### PR DESCRIPTION
**Pull request by rphillips**
Allows for localhost only metrics.

#### What type of PR is this?
/kind feature

#### Does this PR introduce a user-facing change?

```release-note
Add support for --metrics-host.
```

### CODE REVIEW

**Comment by kwilczynski** [comment]
@rphillips, these might be related:

- https://github.com/cri-o/cri-o/issues/7446
- https://github.com/cri-o/cri-o/issues/7448

**Comment by haircommander** [comment]
docs need updating: https://github.com/cri-o/cri-o/actions/runs/7544557013/job/20538161526?pr=7677
I believe there is a `--metrics-port` cli flag, should we have a cli flag for this as well?

**Comment by sohankunkerkar** [comment]
@rphillips could you handle these two failures?
https://github.com/cri-o/cri-o/actions/runs/7560855103/job/20587783493?pr=7677
https://github.com/cri-o/cri-o/actions/runs/7560855084/job/20587869547?pr=7677
code LGTM
/retest

File: docs/crio.8.md [hunk 1]
```diff
@@ -547,4 +550,4 @@ was not joined.
 # SEE ALSO
 
 crio.conf(5), crio.conf.d(5), oci-hooks(5), policy.json(5), registries.conf(5),
-storage.conf(5)
\ No newline at end of file
+storage.conf(5)
```
Comment (haircommander) [inline docs/crio.8.md:553]: to fix this, you can do
```
:set binary
:set noeol
```
assuming you use vim

Comment (rphillips) [reply docs/crio.8.md]: I fixed it progmatically, so it will get generated correctly all the time.

Comment (haircommander) [reply docs/crio.8.md]: wfm!

**Comment by haircommander** [comment]
/approve

LGTM, thanks! @cri-o/cri-o-maintainers PTAL

**Pull request merged**